core/modules/startup/keyboard.js

```diff
@@ -0,0 +1,43 @@
+/*\
+title: $:/core/modules/startup/keyboard.js
+type: application/javascript
+module-type: startup
+
+Keyboard shortcut handling
+
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+// Export name and synchronous status
+exports.name = "global-keyboard";
+exports.platforms = ["browser"];
+exports.after = ["startup"];
+exports.synchronous = true;
+
+exports.startup = function() {
+	if($tw.browser) {
+		$tw.utils.each($tw.wiki.filterTiddlers("[all[shadows+tiddlers]tag[$:/tags/KeyboardShortcut]!has[draft.of]]"),function(title) {
+			var self = this;
+			var key = $tw.wiki.getTiddler(title).fields["key"];
+			$tw.wiki.addEventListener("change",function(changes) {
```

Comment:
The reason I suggested using a change event listener is so that we can dynamically track the active keyboard shortcuts as tiddlers with the required tag come and go. If we only read the shortcuts at startup then the shortcut editor won't work.

So, you'd have a function that reads all the keyboard shortcut tiddlers similar to what you've got here, but you'd run it at startup, and every time you get a change event for one of the tiddlers you're already tracking, or a new tiddler with the required tag.

Reply:
Does this additional setting have a performance impact?

Reply:
ok I'll wrap this whole construct in a change listener